.github/workflows/label-pull-requests.yml

```diff
@@ -0,0 +1,45 @@
+name: Label Pull Request
+on:
+  pull_request_target:
+    types: ['opened', 'reopened', 'synchronize', 'ready_for_review', 'converted_to_draft', 'closed']
+    branches: ['main']
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  labeler:
+    name: 'Apply content-based labels'
+    if: github.event.action == 'opened' || github.event.action == 'reopened' || github.event.action == 'synchronize'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
```

Comment:
Do we really need to checkout? I don't see the checked-out source being used.

Reply:
The labeler action docs say to do it, I assume it is used to determine which files are impacted in some way or another since the PR event does not seem to have that info on it

Reply:
Actually looking again it seems I misunderstood the docs, it should be fine without